OPHD/States/MapViewStateTurn.cpp

```diff
@@ -747,6 +764,12 @@ void MapViewState::nextTurn()
 
 	mMineOperationsWindow.updateTruckAvailability();
 
+	// If this is the first turn with population, then set mTurnNumberOfLanding
+	if (mPopulation.getPopulations().size() > 0 && mTurnCount < mTurnNumberOfLanding)
+	{
+		mTurnNumberOfLanding = mTurnCount;
+	}
```

Comment:
Maybe we should consider setting an additional member field here (which also defaults to `std::numeric_limits<int>::max()`).
```cpp
mMoraleActiveTurn = mTurnNumberOfLanding + gracePeriod[mDifficulty]
```

That means we'd need to do the `gracePeriod` lookup once, when colonists are placed, rather than every turn. 

A new field also means we can simplify checks. Instead of:
```cpp
if (turnsSinceLanding > gracePeriod[mDifficulty])
```
We could instead check:
```cpp
if (mTurnCount > mMoraleActiveTurn)
```
That could allow us to remove the calculation for `turnsSinceLanding`, which was also calculated every turn. It also means you don't need to worry about negative values.

Of course such a new field would also need to be set when loading saved games. It could be calculated in the same way as here, using `mTurnNumberOfLanding`, though only if `mTurnNumberOfLanding` is not `std::numeric_limits<int>::max()`. We would want to avoid overflow in that case.

Reply:
After considering this, I think I would suggest leaving it as it is due to the additional code complexity. Alternatively, we could switch to using `constexpr` arrays to store difficulty-dependent data in a way that doesn't involve a runtime lookup at all. If you still prefer I make this change, I'll see what I can do to keep the complexity down.

Reply:
Hmm, it's curious to hear that. My suggestion was mainly about reducing complexity, though perhaps I phrased it so it sounded like there was an emphasis on efficiency, which was not the intent.

----

Currently the code has:
```cpp
int turnsSinceLanding = mTurnCount - mTurnNumberOfLanding; // If negative, landing has not yet occurred.
```

This is then used a couple of times in a check:
```cpp
if (turnsSinceLanding > gracePeriod[mDifficulty])
```

If we substitute the expression into that check, we get:
```cpp
if (mTurnCount - mTurnNumberOfLanding > gracePeriod[mDifficulty])
```

Re-arranging, we have:
```cpp
if (mTurnCount > mTurnNumberOfLanding + gracePeriod[mDifficulty])
```

The suggestion I made was to store a field:
```cpp
mMoraleActiveTurn = mTurnNumberOfLanding + gracePeriod[mDifficulty]
```

Substituting that into the re-arranged check we get:
```cpp
if (mTurnCount > mMoraleActiveTurn)
```

This seems simpler, at least for the check.

Actually, we never really use `mTurnNumberOfLanding`, other than in the check, which can be more simply written if we had a field for `mMoraleActiveTurn`. I wrote my suggestion as adding a new field, though potentially we could replace one field with the other.

By adjusting the field we store, we can front-load most of the complexity for the check to the point when colonists are first landed.

Storing a different field might mean we change which field gets written to a saved game file. Though the suggestion for the new field field, with a default of `constants::ColonyShipOrbitTime + gracePeriod[mDifficulty]` should behave identically to the current proposal.

Alternatively, we could translate between the current field and the new field suggestion during save/load. That would add some complexity. It wouldn't be a whole lot of extra complexity, though it could be avoided if we settle on a new field before saved games start proliferating with a slightly different field we might then feel the need to continue supporting.

I guess one question here is are there other reasons we'll need to know `mTurnNumberOfLanding`, or is it likely to only ever be used in this one place? If there are other reasons for needing `mTurnNumberOfLanding`, then it may be best to store it as is. If not, using `mMoraleActiveTurn` may allow for simpler code.

Reply:
Given Leeor's comment on Discord about potentially allowing the difficulty setting to be modified during gameplay, perhaps it's best to keep the current design, using the `mTurnNumberOfLanding` as the primary field, and doing the `gracePeriod` lookup each turn.